extensions/odoo_theme/static/style.scss

```
@@ -981,3 +962,19 @@ footer {
 
     }
 }
+
+.o_fullwidth_page {
+    div.toctree-wrapper > ul {
+        display: flex;
+        flex-wrap: wrap;
+        > li {
+            @include media-breakpoint-down(lg) {
+                width: 100%;
+            }
+            @include media-breakpoint-up(lg) {
+                width: 33%;
+            }
+            padding-right: 20px;
+        }
+    }
+}
```

Remove right padding on mobile and on every 3rd column

```suggestion
.o_fullwidth_page {
    div.toctree-wrapper > ul {
        display: flex;
        flex-wrap: wrap;
        > li {
            @include media-breakpoint-down(lg) {
                width: 100%;
            }
            @include media-breakpoint-up(lg) {
                width: 33%;
                &:not(:nth-of-type(3n)){
                    padding-right: $padding-m;
                }
            }
        }
    }
}
```